tasks/vendoring/patches/vendor/tomlkit-dump-inline-table.patch

```diff
@@ -0,0 +1,34 @@
+diff --git a/pipenv/vendor/tomlkit/items.py b/pipenv/vendor/tomlkit/items.py
+index 781e2e98..c3c2d59f 100644
+--- a/pipenv/vendor/tomlkit/items.py
++++ b/pipenv/vendor/tomlkit/items.py
+@@ -21,6 +21,7 @@ if PY2:
+     from pipenv.vendor.backports.functools_lru_cache import lru_cache
+ else:
+     from functools import lru_cache
++from toml.decoder import InlineTableDict
+ 
+ 
+ def item(value, _parent=None):
+@@ -36,7 +37,10 @@ def item(value, _parent=None):
+     elif isinstance(value, float):
+         return Float(value, Trivia(), str(value))
+     elif isinstance(value, dict):
+-        val = Table(Container(), Trivia(), False)
++        if isinstance(value, InlineTableDict):
++            val = InlineTable(Container(), Trivia())
++        else:
++            val = Table(Container(), Trivia(), False)
+         for k, v in sorted(value.items(), key=lambda i: (isinstance(i[1], dict), i[0])):
+             val[k] = item(v, _parent=val)
+ 
+@@ -1000,6 +1004,9 @@ class InlineTable(Item, dict):
+         if key is not None:
+             super(InlineTable, self).__setitem__(key, value)
+ 
++        if hasattr(value, "trivia") and value.trivia.comment:
++            value.trivia.comment = ""
++
+         m = re.match("(?s)^[^ ]*([ ]+).*$", self._trivia.indent)
+         if not m:
+             return
```

Comment:
Is there a bug in TOMLkit, or just a workaround specific to Pipenv? We might want to submit a PR to TOMLkit if it’s the former.

Reply:
You are right, I'll submit a simple fix but I think the owner might have a better one.